### PR TITLE
Flow expiration based on packet time in pcap rather than server time.

### PR DESCRIPTION
**Pull request by mielverkerken**
Fixed by checking on each packet if flows need to be expired based on timestamps in the pcap. We might change this later for performance.

### CODE REVIEW

**Review by Str-Gen — APPROVED** [review]

File: rustiflow/src/main.rs [hunk 1]
```diff
@@ -963,31 +933,11 @@ fn process_packet_ipv4<T>(
...
-            .try_into()
-            .expect("Invalid IP length"),
-    )));
-    let source = std::net::IpAddr::V4(Ipv4Addr::from(u32::from_le_bytes(
-        data.ipv4_destination
-            .to_be_bytes()
-            .try_into()
-            .expect("Invalid IP length"),
-    )));
+    let ts = timestamp.unwrap_or_else(Instant::now);
```
Comment (matissecallewaert) [inline rustiflow/src/main.rs:936]: Nice clarification :rocket: !

File: rustiflow/src/main.rs [hunk 2]
```diff
@@ -1004,54 +954,24 @@ fn process_packet_ipv4<T>(
...
-            )
+            T::new(flow_id.clone(), destination, data.port_destination, source, data.port_source, data.protocol, ts_date)
         }
     });
 
     let end = entry.update_flow(&features, &ts, ts_date, fwd);
-    if end.is_some() {
-        export(&end.unwrap());
+    if let Some(flow) = end {
+        export(&flow);
```
Comment (matissecallewaert) [inline rustiflow/src/main.rs:972]: We should add the same if statement for non contaminant export also.
```rs
            if *NO_CONTAMINANT_FEATURES.lock().unwrap().deref() {
                export(&flow.dump_without_contamination());
            } else {
                export(&flow.dump());
            }
```

**Review by matissecallewaert — CHANGES_REQUESTED** [review]
Saw some inconsistencies, also I think we are using different formatters XD.

File: rustiflow/src/main.rs (hunk 2 above)
Comment (mielverkerken) [reply rustiflow/src/main.rs]: Good catch, fixed!

**Review by matissecallewaert — APPROVED** [review]
Looks good to me as a temporary solution!

**Pull request merged**